torch/csrc/jit/script/compiler.cpp

```diff
@@ -2108,6 +2109,68 @@ struct to_ir {
     } else {
       auto inputs = getNamedValues(apply.inputs(), true);
       auto attributes = emitAttributes(apply.attributes());
+
+      if (auto mv = std::dynamic_pointer_cast<MethodValue>(sv)) {
+        auto& callee = mv->getMethod();
+        try {
+          callee.ensure_defined();
+        } catch (std::exception&) {
```

Comment:
It is important to catch the specific exception. Otherwise this will hide all sorts of bad errors and report them as method recursion.